modules/apps/27-interchain-accounts/keeper/handshake.go

```diff
@@ -81,6 +119,43 @@ func (k Keeper) OnChanOpenTry(
 		return sdkerrors.Wrap(err, "counterparty version validation failed")
 	}
 
+	if portID != types.PortID {
+		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "expected %s, got %s", types.PortID, portID)
+	}
+
+	s := strings.Split(counterparty.PortId, types.Delimiter)
```

Comment:
Maybe we can add the validation check to its own util function just for better readability?

Reply:
Are these the same checks as in `OnChanOpenInit`?

Reply:
> Are these the same checks as in `OnChanOpenInit`?

Essentially yes, but with a subtle difference. In `OnChanOpenInit` the (first-party/self) portID is the encoded string whereas in `OnChanOpenTry` the counterparty portID is encoded with connection sequences. 
As a result in the INIT step, we assert the underlying conn sequence matches index [1] and the counterparty conn sequence matches index [2], however in the TRY step the underlying conn sequence must match index [2] and counterparty conn sequence must match index [1]. 

I've refactored the connection sequence validation to a reusable func `validateConnectionParams`.
Also added two helper funcs to parse controller & host conn sequences from the controller port ID which deals with the swapped indexing